go-controller/pkg/ovn/udn_isolation.go

```diff
@@ -136,3 +141,96 @@ func (oc *DefaultNetworkController) getUDNACLDbIDs(name string, aclDir libovsdbu
+	ops, parseErr, err := oc.setUDNPodOpenPortsOps(podNamespacedName, podAnnotations, lspName, nil)
+	if err != nil {
+		return errors.Join(parseErr, err)
+	}
+	_, err = libovsdbops.TransactAndCheck(oc.nbClient, ops)
+	if err != nil {
+		return utilerrors.Join(parseErr, fmt.Errorf("failed to transact open ports UDN ACLs: %v", err))
+	}
+	return parseErr
+}
+
+// setUDNPodOpenPortsOps returns the operations to add or remove ACLs that allow ingress on required ports.
+// first returned error is parse error, second is db ops error
+func (oc *DefaultNetworkController) setUDNPodOpenPortsOps(podNamespacedName string, podAnnotations map[string]string, lspName string,
+	ops []libovsdb.Operation) ([]libovsdb.Operation, error, error) {
+	udnPGName := libovsdbutil.GetPortGroupName(oc.getSecondaryPodsPortGroupDbIDs())
+
+	ingressMatch, egressMatch, parseErr := getPortsMatches(podAnnotations, lspName)
+	// don't return on parseErr, as we need to cleanup potentially present ACLs from the previous config
+	ingressIDs := oc.getUDNOpenPortDbIDs(podNamespacedName, libovsdbutil.ACLIngress)
+	ingressACL := libovsdbutil.BuildACL(ingressIDs, types.PrimaryUDNAllowPriority,
+		ingressMatch, nbdb.ACLActionAllowRelated, nil, libovsdbutil.LportIngress)
+
+	egressIDs := oc.getUDNOpenPortDbIDs(podNamespacedName, libovsdbutil.ACLEgress)
+	egressACL := libovsdbutil.BuildACL(egressIDs, types.PrimaryUDNAllowPriority,
+		egressMatch, nbdb.ACLActionAllow, nil, libovsdbutil.LportEgress)
+
+	var err error
+	if ingressMatch == "" && egressMatch == "" || parseErr != nil {
+		// no open ports or error parsing annotations, remove ACLs
+		foundACLs, err := libovsdbops.FindACLs(oc.nbClient, []*nbdb.ACL{ingressACL, egressACL})
+		if err != nil {
+			return ops, parseErr, fmt.Errorf("failed to find open ports UDN ACLs: %v", err)
+		}
+		ops, err = libovsdbops.DeleteACLsFromPortGroupOps(oc.nbClient, ops, udnPGName, foundACLs...)
+		if err != nil {
+			return ops, parseErr, fmt.Errorf("failed to remove open ports ACLs from portGroup %s: %v", udnPGName, err)
+		}
+	} else {
+		// update ACLs
+		ops, err = libovsdbops.CreateOrUpdateACLsOps(oc.nbClient, ops, oc.GetSamplingConfig(), ingressACL, egressACL)
+		if err != nil {
+			return ops, parseErr, fmt.Errorf("failed to create or update open ports UDN ACLs: %v", err)
+		}
+
+		ops, err = libovsdbops.AddACLsToPortGroupOps(oc.nbClient, ops, udnPGName, ingressACL, egressACL)
+		if err != nil {
+			return ops, parseErr, fmt.Errorf("failed to add open ports ACLs to portGroup %s: %v", udnPGName, err)
+		}
+	}
+	return ops, parseErr, nil
+}
+
+func (oc *DefaultNetworkController) getUDNOpenPortDbIDs(podName string, aclDir libovsdbutil.ACLDirection) *libovsdbops.DbObjectIDs {
+	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLUDN, oc.controllerName,
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey:      OpenPortACLPrefix + podName,
```

Comment:
should we not have namespace also in the name key? Pod name isnt unique throughout a cluster.

Reply:
yeah bad variable name, in reality namespaced name is used